src/digital_input_stamper.rs

```diff
@@ -0,0 +1,76 @@
+///! Digital Input 0 (DI0) reference clock timestamper
+///!
+///! This module provides a means of timestamping the rising edges of an external reference clock on
+///! the DI0 with a timer value from TIM5.
+///!
+///! # Design
+///! An input capture channel is configured on DI0 and fed into TIM5's capture channel 4. TIM5 is
+///! then run in a free-running mode with a configured tick rate (PSC) and maximum count value
+///! (ARR). Whenever an edge on DI0 triggers, the current TIM5 counter value is captured and
+///! recorded as a timestamp. This timestamp can be either directly read from the timer channel or
+///! can be collected asynchronously via DMA collection.
```

Comment:
If we use DMA to collect the timestamp, wouldn't we also want another DMA transaction to clear the interrupt flag to prevent the overcapture flag from being set? Or would we just ignore the overcapture flag in this scenario?

Reply:
The solution there would likely be to listen for the overcapture flag and generate an interrupt if it ever occurs. Either that, or check the overcapture whenever servicing the completion of the DMA capture.

Reply:
In any case, since we aren't using DMA here, this isn't an issue.